SPECS/tdnf/tdnf.spec

```diff
@@ -167,9 +166,30 @@ if [[ ! -f %{_tdnf_history_db_dir}/history.db ]]; then
     %{_libdir}/tdnf/tdnf-history-util init
 fi
 
+# Make the decision to update yum sym link during post script
+# not package build. Some workloads want to disable tdnf, so
+# decision should not be made during build but during install
+# of package.
+if [ ! -h $(rpm --eval %{_bindir})/yum ]; then
+ln -sf $(rpm --eval %{_bindir})/tdnf $(rpm --eval %{_bindir})/yum
+fi
+
+
+%preun
+# Save the symlink status, this can be removed, once tdnf has moved
+# past 3.5.2-3 as this is needed to help with transition
+readlink /usr/bin/yum > /tmp/tdnf-target
```

Comment:
I personally don't like this, but it was the only way i could ensure backwards comparability. If I add (config) to  /usr/bin/yum, the symlink is still removed, as when the RPM is upgraded, the config flag on the sym link is not registered.

So, i can remove this, if we don't have to retroactively fix the older version

Reply:
I am thinking we can `%config(noreplace)` the yum symlink. Current behavior is that the tdnf package will always update/replace the symlink with the "updated" file. Using the `%config(noreplace)` for the yum symlink, if the file on disk is not touched, then the yum symlink file will replaced with the "updated" yum symlink. But if the file on disk is edited, the edit is preserved (and the updated version is saved in .rpmnew)

Adding @ddstreet for additional thoughts on this

Reply:
Note: I've logged an item to deal with this cleanly in the next major OS release (3.0).

Reply:
For sure using `config(noreplace)` isn't the "right" way to do it, but if it works then it's probably the easiest for mariner 2. Should be tested to make sure it works correctly over various situations (e.g. yum installed and then remove tdnf, yum installed and then upgrade tdnf, upgrade tdnf then install yum, etc).

Reply:
actually on thinking about it, we probably shouldn't use `config(noreplace)`, the yum symlink is just a convenience for users; i think the %post should create the symlink if nothing exists there, and %preun should remove it if it's a symlink to tdnf. No other preun/post logic should be needed.

Reply:
> For sure using `config(noreplace)` isn't the "right" way to do it, but if it works then it's probably the easiest for mariner 2. Should be tested to make sure it works correctly over various situations (e.g. yum installed and then remove tdnf, yum installed and then upgrade tdnf, upgrade tdnf then install yum, etc).

yeah fully agreed on `config(noreplace)` being a hacky workaround for it.

> actually on thinking about it, we probably shouldn't use `config(noreplace)`. Since the yum symlink is just a convenience for users (yum is deprecated and nobody should be using that name anymore) i think the %post should create the symlink if nothing exists there, and %preun should remove it if it's a symlink to tdnf. No other preun/post logic should be needed.

I definitely like this option better

Reply:
OK, removed the transition stage and delete the sym link only if it points to tdnf.